docs/posts/ibis-duckdb-geospatial-dev-guru/index.qmd

```diff
@@ -0,0 +1,180 @@
+date: 2024-01-09
+categories:
+  - blog
+  - duckdb
+  - geospatial
+execute:
+  freeze: false
+---
+
+Spatial Dev Guru wrote a great [tutorial](https://spatial-dev.guru/2023/12/09/geospatial-analysis-using-duckdb/) that walks you through a step-by-step
+geospatial analysis of bike sharing data using DuckDB.
+
+Ibis has support for all the geospatial functions used on the tutorial, and we
+decided to replicate it and share it with you.
+
+## Data
+
+The parquet file used in the original tutorial is available at
+https://github.com/iamgeoknight/common_datasets/tree/main/parquet. The original
+data is also available from the citibike [source](https://s3.amazonaws.com/tripdata/index.html) but as a `.csv` file.
+
+```{python}
+from pathlib import Path
+import tarfile
+from urllib.request import urlretrieve
+
+#Download data
+url = "https://github.com/iamgeoknight/common_datasets/raw/main/parquet/202003-citibike-tripdata.tar.xz"
+tar_path = Path("202003-citibike-tripdata.tar.xz")
+parquet_path = Path("202003-citibike-tripdata.parquet")
+
+if not tar_path.exists():
+    urlretrieve(url, tar_path)
+
+if not parquet_path.exists():
+    with tarfile.open(tar_path, 'r:xz') as t:
+        t.extract("202003-citibike-tripdata.parquet")
+```
+
+Now that we have the data, we import Ibis and turn on the interactive mode, to
+easily explore the output of our queries.
+
+```{python}
+import ibis
+from ibis import _
+from ibis.interactive import *
+```
+
+## Let's get started
+
+Because this dataset does not contain any geometries, we have to load the spatial extension. If the dataset included any geometry columns, Ibis is smart enough to load the extension for us upon reading the data.
+
+```{python}
+con = ibis.duckdb.connect("biketrip.ddb")
+con.load_extension("spatial")
+
+#read data and rename columns to use snake case
+biketrip = con.read_parquet("202003-citibike-tripdata.parquet").rename("snake_case")
+biketrip
+```
+
+
+We have the information about the longitude and latitude for start and end stations,
+to create geometry points and put the spatial features to use.
+
+## Create bike trip table
+
+In the original tutorial, Spatial Dev Guru creates a table with transformed
+"Pickup" and "Dropoff" points. In DuckDB the `st_transform` function takes by default
+points as `YX` (latitude/longitude) while in Ibis,  we assume data in the form
+`XY` (longitude/latitude) to be consistent with PostGIS and Geopandas.
+
+In Ibis the function to tr
+```{python}
+# Notice longitude/latitude order
+pickup = biketrip.start_station_longitude.point(biketrip.start_station_latitude).name("pickup")
+dropoff = biketrip.end_station_longitude.point(biketrip.end_station_latitude).name("dropoff")
+
+#convert is the equivalent of `st_transform`
+biketrip = biketrip.mutate(pickup_point=pickup.convert('EPSG:4326', 'EPSG:3857'),
+                           dropoff_point=dropoff.convert('EPSG:4326', 'EPSG:3857')
+                        )
+
+biketrip[["pickup_point", "dropoff_point"]]
```

Comment:
These line are pretty long. Can you use `_` to condense them a bit? Also should be formatted.